src/api/handlers/twilio/postReply.js

```diff
@@ -1,15 +1,157 @@
+    if (languageConversions[key].includes(reply)) {
+      convertedReply = key;
+      break;
+    }
+  }
 
-  const replyText = TemplatesModel.generateReply(language);
+  return convertedReply;
+};
+
+export async function getAppointments(patientPhoneNumber) {
+  return await db("appointments")
+    .select("appointmentId", "language")
+    .where({ patientPhoneNumber, appointmentIsConfirmed: 0 });
+}
+
+export async function handlePostReply(patientPhoneNumber, messageFromPatient) {
+  let ourResponse;
+
+  if (!messageFromPatient || messageFromPatient === "") {
+    ourResponse = getMessageResponse(
+      "We are sorry, we could not process your response. Please call Sanctuary Refugee Health Centre (Dr. Michael Stephenson) at 226-336-1321"
+    );
+    return ourResponse;
+  }
+
+  const lowercaseMessageFromPatient = messageFromPatient.toLowerCase();
+
+  let appointments;
+  try {
+    appointments = await getAppointments(patientPhoneNumber);
+  } catch (error) {
+    // Can't store reply without appointmentId
+    ourResponse = getMessageResponse(
+      "We are sorry, we could not find an appointment for you. Please call Sanctuary Refugee Health Centre (Dr. Michael Stephenson) at 226-336-1321"
+    );
+    return ourResponse;
+  }
+
+  if (!appointments || appointments.length === 0) {
+    ourResponse = getMessageResponse(
+      "We are sorry, we could not find an appointment for you, or your appointment is already confirmed. Please call Sanctuary Refugee Health Centre (Dr. Michael Stephenson) at 226-336-1321"
+    );
+    return ourResponse;
+  }
+
+  const trx = await db.transaction();
+  const convertedReply = convertReply(lowercaseMessageFromPatient);
 
-  const reply = getMessageResponse(replyText);
+  if (!convertedReply) {
+    ourResponse = getMessageResponse(
+      "We are sorry, we could not confirm your appointment. Please call Sanctuary Refugee Health Centre (Dr. Michael Stephenson) at 226-336-1321"
+    );
+    return ourResponse;
+  }
 
+  try {
+    switch (lowercaseMessageFromPatient) {
+      case "yes":
+        for (const appointment of appointments) {
+          await db("replies")
+            .insert({
+              phoneNumber: patientPhoneNumber,
+              body: messageFromPatient,
+              appointmentId: appointment.appointmentId,
+              time: new Date(),
+            })
+            .transacting(trx);
+          await db("appointments")
+            .where({ appointmentId: appointment.appointmentId })
+            .update({
+              appointmentIsConfirmed: true,
+            })
+            .transacting(trx);
+        }
+        break;
+      case "no":
+        for (const appointment of appointments) {
+          await db("replies")
+            .insert({
+              phoneNumber: patientPhoneNumber,
+              body: messageFromPatient,
+              appointmentId: appointment.appointmentId,
+              time: moment().format("YYYY-MM-DD HH:mm:ss"),
```

Comment:
Should this be `new Date()` instead? to coincide with the case above at line `87`